hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AUtils.java

```diff
@@ -807,10 +742,8 @@ private static AWSCredentialsProvider createAWSV1CredentialProvider(
    * @return the instantiated class
    * @throws IOException on any instantiation failure.
    */
-  private static AwsCredentialsProvider createAWSV2CredentialProvider(
-      Configuration conf,
-      Class<?> credClass,
-      @Nullable URI uri) throws IOException {
+  private static AwsCredentialsProvider createAWSV2CredentialProvider(Configuration conf,
```

Comment:
how about moving this credential stuff into its own class under s3a.auth? its complicated enough to merit it, and pulling it out should make maintenance easier as other changes to S3AUtils won't interfere

Reply:
have moved all of this to new class `AwsCredentialListProvider`. not too sure about the name, happy to change. Also was wondering if all the reflection code (`getInstanceFromReflection()`) in S3AUtils should be moved out too, to this `AwsCredentialListProvider` or maybe another new class in `s3a.auth` ? Didn't move out currently as it's also used by the SignerFactory